Add Support for AWS Lambda Function SDK V2

driverpt opened the pull request:
Fixes #2121

driverpt committed:
Add Support for AWS Lambda Function SDK V2

driverpt committed:
Make Spotless happy

sdelamo commented on aws-sdk-v2/src/main/java/io/micronaut/aws/sdk/v2/service/lambda/LambdaClientFactory.java:
```diff
@@ -0,0 +1,94 @@
+ * limitations under the License.
+ */
+/**
+ * Secrets Manager client factory.
```
correct Java doc

sdelamo commented on aws-sdk-v2/src/main/java/io/micronaut/aws/sdk/v2/service/lambda/package-info.java:
```diff
@@ -0,0 +1,28 @@
+ * limitations under the License.
+ */
+/**
+ * Secrets Manager client factory.
```
Correct Javadoc

sdelamo reviewed (changes requested):
It misses tests

driverpt committed:
Fix compilation issues

driverpt committed:
Initial Commit for Tests

driverpt committed:
Add Lambda Code for Testing with LocalStack

driverpt committed:
Add Integration Tests for Function Client for AWS SDK V2

driverpt committed:
Add missing Function Client Configuration

driverpt committed:
Update Javadoc

sdelamo committed:
Merge branch '4.7.x' into pr/2130

sdelamo commented on aws-sdk-v2/src/main/java/io/micronaut/aws/sdk/v2/service/lambda/LambdaClientFactory.java:
```diff
@@ -0,0 +1,89 @@
+import software.amazon.awssdk.services.lambda.LambdaClientBuilder;
+
+
+public class LambdaClientFactory extends AwsClientFactory<LambdaClientBuilder, LambdaAsyncClientBuilder, LambdaClient, LambdaAsyncClient> {
```
it misses javadoc, `@Factory` and a test.

sdelamo replied:
see: https://github.com/micronaut-projects/micronaut-aws/pull/2130/commits/1149df3d2b21a916f41653ce9cdee6842af32a4b

sdelamo committed:
add javadoc, test and @Factory

sdelamo commented on aws-sdk-v2/src/main/java/io/micronaut/aws/sdk/v2/service/lambda/package-info.java:
```diff
@@ -0,0 +1,28 @@
+/**
+  * Lambda client factory.
+ * @author Luis Duarte
+ * @since 2.6.0
```
wrong since 

```suggestion
 * @since 4.7.0
```

sdelamo commented on aws-sdk-v2/src/main/java/io/micronaut/aws/sdk/v2/service/lambda/package-info.java:
```diff
@@ -0,0 +1,28 @@
+ * limitations under the License.
+ */
+/**
+  * Lambda client factory.
```
```suggestion
 * Lambda client factory.
```

sdelamo committed:
Update aws-sdk-v2/src/main/java/io/micronaut/aws/sdk/v2/service/lambda/package-info.java

sdelamo committed:
Update aws-sdk-v2/src/main/java/io/micronaut/aws/sdk/v2/service/lambda/package-info.java

sdelamo commented on function-client-aws-v2/src/main/java/io/micronaut/function/client/aws/AwsInvokeRequestDefinition.java:
```diff
@@ -0,0 +1,46 @@
+/**
+ * Builds an {@link InvokeRequest} for each definition under {@code aws.lambda.functions}.
+ *
+ * @author graemerocher
```
```suggestion
```

sdelamo commented on function-client-aws-v2/src/main/java/io/micronaut/function/client/aws/AwsInvokeRequestDefinition.java:
```diff
@@ -0,0 +1,46 @@
+ * Builds an {@link InvokeRequest} for each definition under {@code aws.lambda.functions}.
+ *
+ * @author graemerocher
+ * @since 1.0
```
```suggestion
 * @since 4.7.0
```

sdelamo commented on function-client-aws-v2/src/main/java/io/micronaut/function/client/aws/AwsInvokeRequestDefinition.java:
```diff
@@ -0,0 +1,46 @@
+import io.micronaut.function.client.FunctionDefinition;
+
+/**
+ * Builds an {@link InvokeRequest} for each definition under {@code aws.lambda.functions}.
```
is the javadoc correct here?

sdelamo commented on function-client-aws-v2/src/main/java/io/micronaut/function/client/aws/package-info.java:
```diff
@@ -0,0 +1,23 @@
+ * Function AWS lamda configuration.
+ *
+ * @author graemerocher
+ * @since 1.0
```
```suggestion
 * @since 4.7.0
```

sdelamo commented on function-client-aws-v2/src/main/java/io/micronaut/function/client/aws/package-info.java:
```diff
@@ -0,0 +1,23 @@
+/**
+ * Function AWS lamda configuration.
+ *
+ * @author graemerocher
```
```suggestion
```

sdelamo reviewed (changes requested):
It misses documentation

sdelamo committed:
Update function-client-aws-v2/src/main/java/io/micronaut/function/client/aws/package-info.java

sdelamo committed:
Update function-client-aws-v2/src/main/java/io/micronaut/function/client/aws/package-info.java

sdelamo committed:
Update function-client-aws-v2/src/main/java/io/micronaut/function/client/aws/AwsInvokeRequestDefinition.java

sdelamo committed:
Update function-client-aws-v2/src/main/java/io/micronaut/function/client/aws/AwsInvokeRequestDefinition.java

sdelamo committed:
Use AwsInvokeRequestDefinition in javadoc

sdelamo committed:
document Lambda Client

sdelamo committed:
dependencies should be testImplementation

sdelamo committed:
remove testContainers version

sdelamo committed:
doc: how to use function-client-v2

sdelamo committed:
use micronautFunctionClientAwsV2 in test-suite

sdelamo committed:
add logback to src/test/resources

sdelamo committed:
don’t use compress

sdelamo committed:
remove zip

driverpt committed:
Fix lambda code

driverpt committed:
Add AWS SDK IAM to Tests

driverpt committed:
Remove @Internal from Lambda Factory

driverpt committed:
Properly Map Lambda Function Name and extra attributes

driverpt committed:
Fix failing tests

sdelamo committed:
revert v2 dependency

sdelamo committed:
set AwsLambdaFunctionExecutor as @Internal

sdelamo committed:
dont add jackson.databind use JsonMediaTypeCodec

sdelamo committed:
move to package v2

sdelamo committed:
change javadoc

sdelamo committed:
change javadoc

sdelamo committed:
add test

sdelamo committed:
remove @Introspected annotation

sdelamo committed:
remove raw use of parameterized

sdelamo committed:
use BLOCKING instead of IO

sdelamo committed:
remove raw use parameterized

sdelamo committed:
use conversionService

sdelamo committed:
add reactive test

don’t use yaml configuration

sdelamo committed:
fix: run Junit4 test into Junit5 test

This PR moves a Java test which was not being run as it was a JUnit 4 test to `src/test/java` and transforms it into a JUnit5 test.

It updates to latest Micronaut core as it was failing locally for me without this upgrade.

sdelamo committed:
test-suite for function-client-aws v1 and v2

sdelamo committed:
disable binary compatibility

sdelamo committed:
improve docs

sdelamo committed:
add javadoc

sdelamo committed:
fix checkstyle

sdelamo committed:
add javadoc

driverpt commented:
any ETA on this one ?

sdelamo commented:
> any ETA on this one ?

I am going to check again next week. But there were some build failures.

sdelamo committed:
Merge branch '4.7.x' into pr/2130

sdelamo committed:
disabled tests
